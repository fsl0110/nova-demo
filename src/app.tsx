import React, { FC } from "react";
import SwipeableRoutes from "react-swipeable-routes";
import {
  BrowserRouter as Router,
  Route,
  Switch,
  Redirect
} from "react-router-dom";
import { FoodRecallEnforcement, FoodRecallEnforcementD3 } from "./pages";
import { Header, NotFound } from "./components";
import { navLinks } from "./utils/data/navLinks";
import { LinebarChart } from "./components";

export const App: FC = () => (
  <Router>
    <div className="page-wrapper">
      <Header navLinks={navLinks} navLinksAlign="center" />
      <Switch>
        <Route exact path="/" render={() => <Redirect to="/highcharts" />} />
        <SwipeableRoutes enableMouseEvents>
          <Route path="/highcharts" component={FoodRecallEnforcement} />
          <Route path="/d3-js" component={FoodRecallEnforcementD3} />
          <Route path="/linebarchart" component={LinebarChart} />
        </SwipeableRoutes>
        <Route path="*" component={NotFound} />
      </Switch>
    </div>
  </Router>
);
