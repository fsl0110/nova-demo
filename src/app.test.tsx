import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { App } from "./app";

it("renders toMatchSnapshot", () => {
  const mock = shallow(<App />);
  expect(toJson(mock)).toMatchSnapshot();
});
