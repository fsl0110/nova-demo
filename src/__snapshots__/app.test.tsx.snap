// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders toMatchSnapshot 1`] = `
<BrowserRouter>
  <div
    className="page-wrapper"
  >
    <Header
      navLinks={
        Array [
          Object {
            "name": "Highcharts",
            "path": "highcharts",
          },
          Object {
            "name": "D3.js",
            "path": "d3-js",
          },
        ]
      }
      navLinksAlign="center"
    />
    <Switch>
      <Route
        exact={true}
        path="/"
        render={[Function]}
      />
      <withRouter(SwipeableRoutes)
        enableMouseEvents={true}
      >
        <Route
          component={[Function]}
          path="/highcharts"
        />
        <Route
          component={[Function]}
          path="/d3-js"
        />
      </withRouter(SwipeableRoutes)>
      <Route
        component={[Function]}
      />
    </Switch>
  </div>
</BrowserRouter>
`;
